pkg/db/vulnerability_metadata.go

```diff
@@ -7,14 +7,33 @@ type VulnerabilityMetadata struct {
 	Severity     string   // How severe the vulnerability is (valid values are defined by upstream sources currently)
 	Links        []string // URLs to get more information about the vulnerability or advisory
 	Description  string   // Description of the vulnerability
-	CvssV2       *Cvss    // Common Vulnerability Scoring System V2 values
-	CvssV3       *Cvss    // Common Vulnerability Scoring System V3 values
+	Cvss         []Cvss   // Common Vulnerability Scoring System values
 }
 
 // Cvss contains select Common Vulnerability Scoring System fields for a vulnerability.
 type Cvss struct {
-	BaseScore           float64 // Ranges from 0 - 10 and defines for qualities intrinsic to a vulnerability
-	ExploitabilityScore float64 // Indicator of how easy it may be for an attacker to exploit a vulnerability
+	// VendorMetadata captures non-standard CVSS fields that vendors can sometimes
+	// include when providing CVSS information.  This vendor-specific metadata type
+	// allows to capture that data for persisting into the database
+	VendorMetadata interface{}
+	Metrics        CvssMetrics
+	Vector         string // A textual representation of the metric values used to determine the score
+	Version        string // The version of the CVSS spec, for example 2.0, 3.0, or 3.1
+}
+
+type CvssMetrics struct {
+	BaseScore float64 // Ranges from 0 - 10 and defines for qualities intrinsic to a vulnerability
+	// ExploitabilityScore is a pointer to avoid having a 0 value by default.
+	// It is an indicator of how easy it may be for an attacker to exploit
+	// a vulnerability
+	ExploitabilityScore *float64
```

Comment:
As discussed, @wagoodman this is now a pointer, and the helper constructor below takes care of handling the de-referencing

Reply:
awesome! The impact score has the same requirement (is optional) --the base score is alright, that is not optional.

Reply:
Oi. I'll apply the same treatment to `ImpactScore` then. I wasn't aware of that one. This will require updates to the `grype-db-builder` PR too

Reply:
This has been updated